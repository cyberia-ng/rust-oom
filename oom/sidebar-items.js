initSidebarItems({"struct":[["NonEmptyMutSlice","A non-empty mutable slice type, counterpart of `&mut [T]`."],["NonEmptySlice","A non-empty slice type, counterpart of `&[T]`."],["NonEmptyVec","A non-empty vector type, counterpart of `Vec<T>`."]]});