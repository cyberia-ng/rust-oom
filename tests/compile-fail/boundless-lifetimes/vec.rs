use oom::{NonEmptySlice, NonEmptyVec};

fn main() {
    let vec = NonEmptyVec::<u8>::from_vec(vec![0]);
    let bad: NonEmptySlice<'static, _> = vec.as_nonempty_slice();
    drop(vec);
    let _dangling = bad.first();
}
