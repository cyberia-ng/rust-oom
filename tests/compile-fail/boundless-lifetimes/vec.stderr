error[E0107]: wrong number of lifetime arguments: expected 0, found 1
 --> $DIR/vec.rs:5:28
  |
5 |     let bad: NonEmptySlice<'static, _> = vec.as_nonempty_slice();
  |                            ^^^^^^^ unexpected lifetime argument

error[E0599]: no method named `as_nonempty_slice` found for struct `oom::NonEmptyVec<u8>` in the current scope
 --> $DIR/vec.rs:5:46
  |
5 |     let bad: NonEmptySlice<'static, _> = vec.as_nonempty_slice();
  |                                              ^^^^^^^^^^^^^^^^^ method not found in `oom::NonEmptyVec<u8>`
